app/services/sift/ApplicationSiftService.scala

```diff
@@ -34,21 +36,28 @@ import scala.language.postfixOps
 object ApplicationSiftService extends ApplicationSiftService {
   val applicationSiftRepo: ApplicationSiftMongoRepository = repositories.applicationSiftRepository
   val applicationRepo: GeneralApplicationMongoRepository = repositories.applicationRepository
+  val contactDetailsRepo = repositories.faststreamContactDetailsRepository
   val schemeRepo = SchemeYamlRepository
   val dateTimeFactory = DateTimeFactory
+  val emailClient = CSREmailClient
 }
 
 trait ApplicationSiftService extends CurrentSchemeStatusHelper with CommonBSONDocuments {
-  def applicationSiftRepo: ApplicationSiftRepository
 
+  def applicationSiftRepo: ApplicationSiftRepository
   def applicationRepo: GeneralApplicationRepository
-
+  def contactDetailsRepo: ContactDetailsRepository
   def schemeRepo: SchemeRepository
+  def emailClient: CSREmailClient
 
   def nextApplicationsReadyForSiftStage(batchSize: Int): Future[Seq[ApplicationForSift]] = {
     applicationSiftRepo.nextApplicationsForSiftStage(batchSize)
   }
 
+  def processNextApplicationFailedAtSift: Future[Unit] = applicationSiftRepo.nextApplicationFailedAtSift.flatMap(_.map { application =>
+    applicationRepo.addProgressStatusAndUpdateAppStatus(application.applicationId, ProgressStatuses.FAILED_AT_SIFT)
+  }.getOrElse(Future()))
```

Comment:
should it be `Future.success` ?

Reply:
This will be a Future success as there is no exception, just a unit value.

Reply:
Future.success returns already completed future, while Future() - create another empty thread and execute it. Few extra cpu cycles, Do we need them?

Reply:
👍  TIL.  Will change